### PR TITLE
Account and limit number of lids per file to reduce spkies during com…

### PR DESCRIPTION
**Pull request by baldersheim**
…paction.

This is to avoid ending up with a very long listof lids that are removed but not compacted away due to lidspace compaction being disabled when
removes and delets are too frequent.

@toregge and @geirst  PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst @toregge ping

File: searchlib/src/tests/docstore/logdatastore/logdatastore_test.cpp [hunk 1]
```diff
@@ -202,15 +202,24 @@ TEST("test that DirectIOPadding works accordng to spec") {
...
-TEST("testGrowing") {
-    FastOS_File::EmptyAndRemoveDirectory("growing");
-    EXPECT_TRUE(FastOS_File::MakeDirectory("growing"));
-    LogDataStore::Config config; //(100000, 0.1, 3.0, 0.2, 8, true, CompressionConfig::LZ4,
-                                // WriteableFileChunk::Config(CompressionConfig(CompressionConfig::LZ4, 9, 60), 1000));
-    config.setMaxFileSize(100000).setMaxDiskBloatFactor(0.1).setMaxBucketSpread(3.0).setMinFileSizeFactor(0.2)
-            .compactCompression({CompressionConfig::LZ4})
-            .setFileConfig({{CompressionConfig::LZ4, 9, 60}, 1000});
-    vespalib::ThreadStackExecutor executor(8, 128*1024);
+class TmpDirectory {
```
Comment (geirst) [inline searchlib/src/tests/docstore/logdatastore/logdatastore_test.cpp:205]: Consider using search::test::DirectoryHandler (searchlib/test/directory_handler.h) instead.

**Review by geirst — APPROVED** [review]
👍 @toregge please also have a look

**Review by toregge — APPROVED** [review]

File: searchlib/src/tests/docstore/logdatastore/logdatastore_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/docstore/logdatastore/logdatastore_test.cpp]: Done

**Pull request merged**